src/python/turicreate/test/test_api_visibility.py

```diff
@@ -0,0 +1,363 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2017 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+from __future__ import print_function as _
+from __future__ import division as _
+from __future__ import absolute_import as _
+import unittest
+import turicreate
+import inspect
+
+MODULES = ['clustering', 'data_structures', 'graph_analytics', 'recommender']
+CLUSTER = ['kmeans', 'dbscan']
+RECOMMENDERS = ['popularity_recommender',
+                'item_similarity_recommender',
+                'factorization_recommender',
+                'ranking_factorization_recommender',
+                'item_content_recommender']
+DATA_STRUCTURES = ['SFrame', 'SArray', 'SGraph', 'Vertex', 'Edge']
+GRAPH_ANALYTICS = ['connected_components', 'graph_coloring', 'kcore', 'load_sgraph', 'pagerank',
+                   'triangle_counting',  'shortest_path']
+GENERAL = ['load_model', 'load_sframe', 'load_sarray']
+
+class TuriTests(unittest.TestCase):
+    def test_top_level(self):
+        for x in (MODULES + CLUSTER + DATA_STRUCTURES + GRAPH_ANALYTICS +
+                  GENERAL + RECOMMENDERS):
+            self.assertTrue(x in dir(turicreate))
+
+        # TODO Test whether or not things are NOT visible
+
+def load_lazy_modules():
```

Comment:
lazy_modules_force_load might be a better name. ;-)

Reply:
I'll give this one to you.

Reply:
Done.